### DISCUSSION TITLE
HA automations failing due to controller reads

### DISCUSSION THREAD

**Discussion by pyriel800** [davidrapan/ha-solarman#144] [General]
First up, I'm loving this work and thanks David!

When running integrations against timed events: many of us in the UK use Octopus on the Agile tariff so might change the time of use switch every 30 minutes depending on the price in that slot.

I've noticed that occassionally I get an automation step failure (yes this is a very simple automation but this is all it needs to do!):
![image](https://github.com/user-attachments/assets/f8041ab3-fb3b-48e1-8fe5-f25bf0ff8fb4)

I've change the retries, as I expect that will fix it/work around the read/write clash - BUT I wonder if there's some sort of defensive logic that could be applied to mutex the read process from the execution of the write.

I'm hoping someone will tell me I'm an idiot and I should do XYZ but sharing here anyway!

**Comment by davidrapan** [comment]
If you want to use it in automation and be 99.99% sure it will succeed set `wait_for_attempts` to at least 30.

**Comment by gedger** [comment]
Another option would be to add it to your inverter specification as a platform number and then you can change the number in HA and let the solarman integration write the value. Here's an example for my inverter. 

      # This limits the max power output of the inverter as a % of the total
      - name: "Power Limit"
        code: 0x03
        platform: "number"
        uom: "%"
        digits: 2
        rule: 1
        scale: 0.01
        registers: [3051]
